File: 1903. Largest Odd Number in String/Solution.cpp
// Author        : tarlansoltanov
// Title         : Largest Odd Number in String
// ID            : 1903
// URL           : https://leetcode.com/problems/largest-odd-number-in-string/
// Difficulty    : Easy
// Language      : C++


class Solution
{
    public:
        string largestOddNumber(string num)
        {
            int n = num.length();

            while(n--)
                if (num[n] == '1' || num[n] == '3' || num[n] == '5' || num[n] == '7' || num[n] == '9')
                    return num.substr(0, n + 1);

            return "";
        }
};

File: 1903. Largest Odd Number in String/Solution.py
# Author        : tarlansoltanov
# Title         : Largest Odd Number in String
# ID            : 1903
# URL           : https://leetcode.com/problems/largest-odd-number-in-string/
# Difficulty    : Easy
# Language      : Python


class Solution:
    def largestOddNumber(self, num: str) -> str:
        for i in range(len(num) - 1, -1, -1):
            if int(num[i]) % 2 == 1:
                return num[0:i+1]
        return ""
        